### PR TITLE
Cython

### PR DESCRIPTION
**Pull request by rcooke-ast**
Removed a bunch of deprecated cython functions. I've also removed a bunch of cython functions that are not used by default during a standard reduction.

At this point, there are about 40 cython functions remaining, which are mostly used in edge tracing (`artrace.py`) and frame combination (`arcomb.py`) routines.

I have checked that the development suite is unbroken.

### CODE REVIEW

File: pypit/arparse.py [hunk 1]
```diff
@@ -1002,15 +1002,14 @@ def reduce_flatfield_method(self, v):
         v : str
           value of the keyword argument given by the name of this function
         """
-        allowed = ['polyscan', 'bspline']
+        allowed = ['bspline']
```
Comment (profxj) [inline pypit/arparse.py]: Is there a chance that people had been using polyscan, i.e.
this is not backwards compatible?

**Comment by profxj** [comment]
Only 1 comment from me on backwards compatibility.

File: pypit/arparse.py (hunk 1 above)
Comment (rcooke-ast) [inline pypit/arparse.py]: My feeling is that the bspline does a better job, and there was a bug with the polyscan algorithm near the edges. So, if people did use it in the past, it's probably best that they don't anymore...

Comment (profxj) [inline pypit/arparse.py]: Should we raise a deprecation warning/error if they try, 
and an advisement to use the bspline instead?

Comment (rcooke-ast) [inline pypit/arparse.py]: Good idea - I'll make it so.

Comment (rcooke-ast) [inline pypit/arparse.py]: I've added a new function called "check_deprecated"

**Comment by profxj** [comment]
Merging.

**Pull request merged**